app/src/main/java/com/babylon/wallet/android/presentation/settings/accountsecurity/ledgerhardwarewallets/LedgerHardwareWalletsViewModel.kt

```diff
@@ -77,18 +71,16 @@ class LedgerHardwareWalletsViewModel @Inject constructor(
 
     fun disableAddLedgerButtonUntilConnectionIsEstablished() {
         _state.update {
-            it.copy(
-                showContent = LedgerHardwareWalletsUiState.ShowContent.Details,
-                isAddLedgerButtonEnabled = true
-            )
+            it.copy(showContent = LedgerHardwareWalletsUiState.ShowContent.Details)
         }
-        viewModelScope.launch {
-            ledgerMessenger.isConnected.filter { it }.firstOrNull()?.let {
+        ledgerMessenger.isAnyLinkedConnectorConnected
+            .dropWhile { isConnected ->
                 _state.update { state ->
-                    state.copy(isAddLedgerButtonEnabled = false)
+                    state.copy(isNewLinkedConnectorConnected = isConnected)
                 }
+                isConnected.not() // continue while isConnected is not true
             }
-        }
+            .launchIn(viewModelScope)
```

Comment:
the previous version didn't actually work so I fixed it.

But what I want to point out here is how many amazing operators Flow has! When you find yourself doing some workarounds with Flows you should start looking into the available operators! Most probably you will find a much more elegant solution.

For example the `dropWhile` fits perfectly in cases we want to observe the progress of a task, and when the task is done the collection stops! This is exactly what I am doing there. The screen collecting the (just added) link connector state until it gets the isConnected = true. 
The `isNewLinkedConnectorConnected` as long as is false it shows the loading spinner on the button